### DISCUSSION TITLE
Wix v4 HarvestProject not harvesting project references

### DISCUSSION THREAD

**Discussion by ian-g-holm-intel** [wixtoolset/issues#7471] [Questions]
Can anyone tell me how to get HarvestProject to also harvest my project's references? Not matter what I do, my primary project's references do not get harvested.

Here is an example github project that reproduces this issue.
https://github.com/Retik/HarvestExample

**Comment by MsnVt** [comment]
### Harvest Element:
You might need to add a Harvest element to your property group:
```
<ItemGroup>
  <ProjectReference Include="MyProject.csproj" Harvest="true" … DoNotHarvest="true" />
</ItemGroup>

```

### Version inside Package Reference:
 I noticed you didn't include a version inside your package reference either: 
 ```
 <ItemGroup>
        <PackageReference Include="WixToolset.Util.wixext" />
        <PackageReference Include="WixToolset.Heat" />
    </ItemGroup>
 ```
 Maybe changing it to look a bit more like this will help you:
 ```
  <ItemGroup>
        <PackageReference Include="WixToolset.Util.wixext" Version="<Version_Number>"/>
        <PackageReference Include="WixToolset.Heat" Version="<Version_Number>"/>
    </ItemGroup>
 ```

**Misc.** 
In addition, it looks like you also have an extra parentheses within your the `Condition` of your `PropertyGroup `below:
```
<PropertyGroup Condition=" '$(Configuration))' == 'Debug' ">
        <DebugType>full</DebugType>
        <Optimize>false</Optimize>
    </PropertyGroup>
```


ref:[wix:heat](https://wixtoolset.org/docs/tools/heat/)
wix3: [wix:v3:harvestprojects](https://wixtoolset.org/docs/v3/msbuild/target_reference/harvestprojects/)
relevant and very recent StackOverflow question on how the WiX 4 Harvest Directory works: [Explanation For WiX 4 Harvest Directory?](https://stackoverflow.com/questions/76056261/explanation-for-wix-4-harvest-directory)

_Full Disclosure, I'm more used to working with version 3 of the WiX toolset, so this advice might have been found in haste of trying to help you_

**Reply by ian-g-holm-intel** [reply]
I'm following the documentation for HarvestProject here, https://wixtoolset.org/docs/tools/heat/#using-harvestproject-to-harvest-output-from-a-project

When using HarvestProject, I don't think the ProjectReference should have Harvest="true". 

My PackageReferences do not have versions specified because we're using ManagePackageVersionsCentrally and specifying them in a Directory.Packages.props file. The extra parenthesis was a mistake, but not the cause of my issues.

**Comment by ian-g-holm-intel** [comment]
I've updated the description with a link to a github project that reproduces this issue. It includes 3 parojects, ExampleApp, ExampleLibrary and ExampleMsi. The ExampleMsi harvests ExampleApp and ExampleApp references ExampleLibrary. The HarvestProject is only harvesting ExampleApp.exe but not ExampleLibrary.dll.
